tests/queries/0_stateless/02497_storage_file_reader_selection.sh

```diff
@@ -7,7 +7,7 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 DATA_FILE=test_02497_$CLICKHOUSE_TEST_UNIQUE_NAME.tsv
 echo -e 'key\nfoo\nbar' > $DATA_FILE
 
-$CLICKHOUSE_LOCAL --storage_file_read_method=mmap --print-profile-events -q "SELECT * FROM file($DATA_FILE) FORMAT Null" 2>&1 | grep -F -c "CreatedReadBufferMMap"
+$CLICKHOUSE_LOCAL --storage_file_read_method=mmap --print-profile-events -q "SELECT * FROM file($DATA_FILE) FORMAT Null" 2>&1 | grep -F -c "CreatedReadBufferMMap:"
```

Comment:
AFAIU sometimes this query can get both `CreatedReadBufferMMap` and `CreatedReadBufferMMapFailed` events . Let's grep only for `CreatedReadBufferMMap`

Reply:
This does not helps completely - https://s3.amazonaws.com/clickhouse-test-reports/52068/b6f9be2ad842a4cb9338d521bab07b931a0b3271/fast_test.html

I think that there is some problems with profile events, but I cannot reproduce it locally, either by slowing down the process to 0.1% of cpus artificially and without.

BTW in which case it may get both events?

Reply:
> BTW in which case it may get both events?

Actually, I didn't fully understand it and it was more like a hypothesis, I could be wrong